### PR TITLE
Feat: add password reset screens

### LINKED ISSUE #26: Feat: forgot & reset password screens designs
#### Description

A user of the app might forget their password. In that case they need a place where they can reset their password in a secure way.

#### Tasks

- [x] Design forgot password screen
- [x] Design OTP code verification screen
- [x] Design create new password screen 
- [x] Design password reset success screen

#### Developer Notes

Since the styling framework is not already setup i will be using the usual css and this will be refactored once it is merged in the `dev` branch.

#### Related Issues

- N/A

#### Definition of Done

- All screens looks and behaves as the designs
- Navigation between screens works perfectly

### PR DESCRIPTION
**Pull request by fredshema**
#### What does this PR do?

This PR includes designs of the forgot & reset password screens

#### Description of Task to be completed?

- Add reset password, otp verification, and new password screens
- Add navigation header component
- Add icon component
- Add typed routes configuration
- Add necessary props on button component
- Add input component

#### How should this be manually tested?

On the first page there is a link with text `Reset Password`, click it and it will redirect you to the reset password screen. Follow the continue buttons to check all screens.

#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker/Trello stories?

N/A

#### Questions:

N/A

### CODE REVIEW

**Review by Emmyfrank — APPROVED** [review]

**Review by chrysloi — COMMENTED** [review]
Please remove yarn.lock as we agreed on using only npm.

**Review by Emmyfrank — APPROVED** [review]

**Review by chrysloi — APPROVED** [review]
🤝🏾

**Review by 911samuel — APPROVED** [review]

File: app/_layout.tsx
Comment (Adadie) [inline app/_layout.tsx:1]: We already have the _layout file on dev. Rebase on the current changes

**Review by Adadie — CHANGES_REQUESTED** [review]
Also squash your commits into one

**Review by Emmyfrank — APPROVED** [review]

**Comment by fredshema** [comment]
I updated the expo SDK to version 51 because it was failing to run on device. Version 50 is running in a simulator only after they updated the expo app and I can't install an earlier version of the expo app.

**Review by justinmihigo — APPROVED** [review]

**Review by Latiah — APPROVED** [review]

**Review by Adadie — APPROVED** [review]
Moving forward let's use native-wind styles

**Pull request merged**